Module#02/service/src/main/java/com/epam/esm/mapper/CertificateMapper.java

```diff
@@ -0,0 +1,91 @@
+package com.epam.esm.mapper;
+
+import com.epam.esm.domain.Certificate;
+import com.epam.esm.domain.Tag;
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.dto.CertificateWithoutTagDto;
+import com.epam.esm.dto.TagDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toSet;
+
+@Slf4j
+@Component
+public final class CertificateMapper implements EntityMapper<Certificate, CertificateDto> {
```

Comment:
Please use MapStruct features: https://www.baeldung.com/mapstruct

Reply:
MapStruct logic was changed, toDtos, to Entities were added